test/extended/node_tuning/node_tuning_utils.go

```diff
@@ -0,0 +1,215 @@
+}
+
+func findCondition(conditions []configv1.ClusterOperatorStatusCondition, name configv1.ClusterStatusConditionType) *configv1.ClusterOperatorStatusCondition {
+	for i := range conditions {
+		if name == conditions[i].Type {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+// try to check if CO status is available with multiple retry, the maxinum is 5 times, re-try one time each minutes.
+// just in case the co status become degraded or processing during ocp upgrade scenario or node reboot/scale out.
+func assertIfSpecifiedCOWithAvailableStateWithRetry(oc *exutil.CLI, coName string) error {
+	errWait := wait.Poll(1*time.Minute, 5*time.Minute, func() (bool, error) {
+		isCOAvailable, err := isCOAvailableState(oc, coName)
+		if err != nil {
+			return false, err
+		}
+		if isCOAvailable {
+			e2e.Logf("the status of cluster operator %v keep on available state", coName)
+			return true, nil
+		}
+		e2e.Logf("the status of co %v doesn't stay on expected state, will check again", coName)
+		return false, nil
+	})
+	return errWait
+}
+
+// isCOAvailable returns true when the ClusterOperator's coName status conditions are as follows: Available true, Progressing false and Degraded false.
+func isCOAvailableState(oc *exutil.CLI, coName string) (bool, error) {
+	var (
+		clusterOperators []configv1.ClusterOperator
+		co               configv1.ClusterOperator
+		isAvailable      bool
+	)
+	clusterOperatorsList, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		e2e.Logf("fail to get cluster operator list with error")
+		return false, err
+	}
+	clusterOperators = clusterOperatorsList.Items
+	for _, clusterOperator := range clusterOperators {
+		if clusterOperator.Name == coName {
+			co = clusterOperator
+			e2e.Logf("co name is %v", co.Name)
+			e2e.Logf("desiredCO.status.conditions of %v is %v", co.Name, co.Status.Conditions)
+			break
+		}
+	}
+	available := findCondition(co.Status.Conditions, configv1.OperatorAvailable)
+	degraded := findCondition(co.Status.Conditions, configv1.OperatorDegraded)
+	progressing := findCondition(co.Status.Conditions, configv1.OperatorProgressing)
+	e2e.Logf("the status of co %v is available.Status [%v] degraded.Status [%v] and progressing.Status [%v]", coName, available.Status, degraded.Status, progressing.Status)
+	if available.Status == configv1.ConditionTrue &&
+		degraded.Status == configv1.ConditionFalse &&
+		progressing.Status == configv1.ConditionFalse {
+		isAvailable = true
+	}
+	e2e.Logf("co/%v status is %v", coName, co.Status)
+	return isAvailable, nil
+}
+
+// retry to check if mcp master/worker mcp is on updated state, the maxinum is 10 times, re-try one time each minutes
+// just in case the mcp state become updating during ocp upgrade or node reboot/scale out or other test case change mcp
+func assertIfSpecifiedMCPWithUpdatedStateWithRetry(mcps dynamic.NamespaceableResourceInterface, mcpName string) error {
```

Comment:
Thank you for the explanation. In that case, if you agree, this function should probably called something like: waitForUpdatedMCP() .  In my view, that would help people to understand it better.